tests/app.test.ts

```diff
@@ -103,4 +103,53 @@ describe('Application Integration Tests', () => {
                 expect(e.response.status).toBe(400);
             });
     });
+
+    it('uploads a manuscript file', async () => {
+        const body = new FormData();
+
+        const loginResponse = await axios.post(
+            'http://localhost:3000/graphql',
+            {
+                query: `
+                    mutation StartSubmission($articleType: String!) {
+                        startSubmission(articleType: $articleType) {
+                            id
+                        }
+                    }
+                `,
+                variables: {
+                    articleType: 'researchArticle',
+                },
+            },
+            { headers: { Authorization: `Bearer ${jwtToken}` } },
+        );
+
+
+        const id = loginResponse.data.data.startSubmission.id;
+        const query = `mutation UploadManuscript($id: ID!, $file: Upload!, $fileSize: Int!) {
+            uploadManuscript(id: $id, file: $file, fileSize: $fileSize) {
+                id
+            }
+        }`;
+
+        const operations = {
+            query,
+            variables: {
+                id,
+                file: null,
+                fileSize: 2,
+            },
+        };
+
+        body.append('operations', JSON.stringify(operations));
+        body.append('map', '{ "1": ["variables.file"] }');
+        body.append('1', 'a', { filename: 'a.txt' });
```

Comment:
This looks like an odd way to build the request body, how have we made sure this has the same shape as an Apollo client mutation call?

Reply:
If i remember correctly this is how you need to do it for graphql. @norisalsaadie  can you chime in ?

Reply:
I'm afraid it's correct. As per the spec https://github.com/jaydenseric/graphql-multipart-request-spec#curl-request